.eslintrc

```diff
@@ -1,14 +1,49 @@
 {
-    "extends": "eslint:recommended",
+    "extends": [
+      "eslint:recommended",
+      "plugin:@typescript-eslint/eslint-recommended",
+      "plugin:@typescript-eslint/recommended",
+      "plugin:react/recommended"
+    ],
     "env": {
       "amd": true,
       "node": true,
       "es6": true,
     },
+    "settings": {
+      "react": {
+        "version": "detect", // React version. "detect" automatically picks the version you have installed.
+      }
+    },
     "rules": {
-        "semi": 2,
-        "eol-last": 2,
-        "eqeqeq": "error",
-        "no-console": 0
+        "max-len": ["warn", 120, 2],
+        "indent": "off",
+        "@typescript-eslint/indent": ["error", 2],
+        "@typescript-eslint/prefer-for-of": "warn",
```

Comment:
why "warn"? Do we not want to enforce these rules?

Reply:
will gradually turn these all into "error"